src/main/java/com/google/maps/StaticMapsRequest.java

```diff
@@ -0,0 +1,415 @@
+        throw new IllegalArgumentException(
+            "Color '" + color + "' doesn't match acceptable color pattern.");
+      }
+
+      this.color = color;
+    }
+
+    private static final Pattern labelPattern = Pattern.compile("^[A-Z0-9]$");
+
+    /** Specifies a single uppercase alphanumeric character from the set {A-Z, 0-9}. */
+    public void label(String label) {
+      if (!labelPattern.matcher(label).matches()) {
+        throw new IllegalArgumentException(
+            "Label '" + label + "' doesn't match acceptable label pattern.");
+      }
+
+      this.label = label;
+    }
+
+    /**
+     * Set a custom icon for these markers.
+     *
+     * @param url URL for the custom icon.
+     * @param anchorPoint The anchor point for this custom icon.
+     */
+    public void customIcon(String url, CustomIconAnchor anchorPoint) {
+      this.customIconURL = url;
+      this.anchorPoint = anchorPoint;
+    }
+
+    /** Add the location of a marker. At least one is required. */
+    public void addLocation(String location) {
+      locations.add(location);
+    }
+
+    /** Add the location of a marker. At least one is required. */
+    public void addLocation(LatLng location) {
+      locations.add(location.toUrlValue());
+    }
+
+    @Override
+    public String toUrlValue() {
+      List<String> urlParts = new ArrayList<>();
+
+      if (customIconURL != null && anchorPoint != null) {
+        urlParts.add("icon:" + customIconURL);
+        urlParts.add("anchor:" + anchorPoint.toUrlValue());
+      }
+
+      if (size != null || size != MarkersSize.normal) {
+        urlParts.add("size:" + size.toUrlValue());
+      }
+
+      if (color != null) {
+        urlParts.add("color:" + color);
+      }
+
+      if (label != null) {
+        urlParts.add("label:" + label);
+      }
+
+      urlParts.addAll(locations);
+
+      return String.join("|", urlParts);
+    }
+  }
+
+  /**
+   * <code>markers</code> parameter defines a set of one or more markers (map pins) at a set of
+   * locations. Each marker defined within a single markers declaration must exhibit the same visual
+   * style; if you wish to display markers with different styles, you will need to supply multiple
+   * markers parameters with separate style information.
+   *
+   * @param markers A group of markers with the same style.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest markers(Markers markers) {
+    return param("markers", markers);
+  }
+
+  public static class Path implements UrlValue {
+
+    private Integer weight;
+    private String color;
+    private String fillcolor;
+    private boolean geodesic;
+    private List<String> points = new ArrayList<>();
+
+    /**
+     * Specifies the thickness of the path in pixels. If no weight parameter is set, the path will
+     * appear in its default thickness (5 pixels).
+     */
+    public void weight(int weight) {
+      this.weight = weight;
```

Comment:
should setting this to zero clear the Integer?

Reply:
Good idea. Implemented in `toUrlValue()`